File: rectangle/Rectangle.php
<?php
	class Rectangle { 
		private $width, $height; 
	
		function __construct($width, $height) { 
			$this->width  = $width; 
			$this->height = $height; 
		}

		function CalcArea() { 
			return $this->width * $this->height; 
		}

		function CalcDiagonal() { 
			$a = $this->width  * $this->width; 
			$b = $this->height * $this->height; 
			$c = $a + $b; 
			return sqrt($c); 
		} 
	}
?>

File: rectangle/ruby/Rectangle.rb
class Rectangle
	def initialize(w, h) 
		@width = w
		@height = h 
	end

	def CalcArea 
		@width * @height;
	end

	def CalcDiagonal
		a = @width * @width 
		b = @height * @height 
		c = a + b

		Math.sqrt(c) # returns the square root, despite having no "return" statement
	end	
end
